opensrp-giz-malawi/src/main/java/org/smartregister/giz/dao/ReportDao.java

```diff
@@ -0,0 +1,421 @@
+import org.smartregister.giz.util.GizConstants;
+import org.smartregister.immunization.db.VaccineRepo;
+import org.smartregister.immunization.domain.Vaccine;
+import org.smartregister.immunization.domain.VaccineCondition;
+import org.smartregister.immunization.domain.VaccineSchedule;
+import org.smartregister.immunization.domain.VaccineTrigger;
+import org.smartregister.immunization.domain.jsonmapping.Condition;
+import org.smartregister.immunization.domain.jsonmapping.Due;
+import org.smartregister.immunization.domain.jsonmapping.Expiry;
+import org.smartregister.immunization.domain.jsonmapping.Schedule;
+import org.smartregister.immunization.domain.jsonmapping.VaccineGroup;
+import org.smartregister.immunization.repository.VaccineRepository;
+import org.smartregister.immunization.util.VaccinatorUtils;
+import org.smartregister.repository.EventClientRepository;
+import org.smartregister.util.AssetHandler;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import timber.log.Timber;
+
+public class ReportDao extends AbstractDao {
+
+    private static HashMap<String, HashMap<String, VaccineSchedule>> vaccineSchedules;
+    @SuppressLint("StaticFieldLeak")
+    private static Context context;
+    private static final SimpleDateFormat sdf = new SimpleDateFormat(GizConstants.DateTimeFormat.yyyy_MM_dd, Locale.US);
+
+
+    private static void readLocations(TreeNode<String, Location> node, Map<String, String> locations) {
+        locations.put(node.getId(), node.getLabel());
+        if (node.getChildren() != null) {
+            for (Map.Entry<String, TreeNode<String, Location>> entry : node.getChildren().entrySet()) {
+                readLocations(entry.getValue(), locations);
+            }
+        }
+    }
+
+    @NonNull
+    public static Map<String, String> extractRecordedLocations() {
+        return extractRecordedLocations(CoreLibrary.getInstance().context().anmLocationController().get());
+    }
+
+    @NonNull
+    public static Map<String, String> extractRecordedLocations(String locationData) {
+        Map<String, String> defaultTree = new HashMap<>();
+        LocationTree locationTree = AssetHandler.jsonStringToJava(locationData, LocationTree.class);
+        if (locationTree != null) {
+            LinkedHashMap<String, TreeNode<String, Location>> locationHierarchyMap = locationTree.getLocationsHierarchy();
+            for (Map.Entry<String, TreeNode<String, Location>> entry : locationHierarchyMap.entrySet()) {
+                readLocations(entry.getValue(), defaultTree);
+            }
+        }
+
+        Map<String, String> locations = new HashMap<>();
+        String sql = "SELECT DISTINCT location_id FROM ec_family_member_location";
```

Comment:
Can this be limited to clients who are children ... or that does not apply here?

Reply:
That doesn't apply here